pkg/grizzly/resources.go

```diff
@@ -30,6 +30,13 @@ func (ref ResourceRef) String() string {
 // Resource represents a single Resource destined for a single endpoint
 type Resource struct {
 	Body map[string]interface{}
+
+	Provenance struct {
+		Format     string
+		Location   string
+		Path       string
+		Rewritable bool
```

Comment:
This seems to be always dependent on the format, is storing this even necessary? Couldn't a "writer" decide whether to actually write depending on the format?

Reply:
At first, a json resource with multiple objects wouldn't be writable. But you're saying decide at writing time not at parse time.